src/main/java/com/xjeffrose/xio/http/RequestBuffer.java

```diff
@@ -0,0 +1,92 @@
+package com.xjeffrose.xio.http;
+
+import com.xjeffrose.xio.client.ClientState;
+import com.xjeffrose.xio.client.DefaultChannelInitializer;
+import com.xjeffrose.xio.client.XioClient;
+import com.xjeffrose.xio.client.XioClientBootstrap;
+import com.xjeffrose.xio.server.Route;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPromise;
+import io.netty.handler.codec.http2.Http2DataFrame;
+import io.netty.handler.codec.http2.Http2Headers;
+import io.netty.util.AttributeKey;
+import io.netty.util.ReferenceCountUtil;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import java.util.function.BiFunction;
+
+@Slf4j
+public class RequestBuffer extends ChannelDuplexHandler {
+
+  public static class WriteReady {
+    public static final WriteReady INSTANCE = new WriteReady();
+  }
+
+  public static class ObjectAndPromise {
+    public final Object object;
+    public final ChannelPromise promise;
+
+    ObjectAndPromise(Object object, ChannelPromise promise) {
+      this.object = object;
+      this.promise = promise;
+    }
+
+    public ChannelFuture apply(BiFunction<Object, ChannelPromise, ChannelFuture> write) {
+      return write.apply(object, promise);
+    }
+  }
+
+  List<ObjectAndPromise> writeBuffer = new ArrayList<>();
```

Comment:
Any reason for this not to be `private final`?

Reply:
Fixed.